core/src/main/kotlin/io/fritz2/binding/validation.kt

```diff
@@ -26,13 +29,16 @@ abstract class Validator<D, M: WithSeverity, T> {
 
     abstract fun validate(data: D, metadata: T): List<M>
 
-    fun isValid(msg: M): Boolean {
+    open fun validPredicate(msg: M): Boolean {
         return msg.severity < Severity.Error
     }
 
     fun isValid(): Flow<Boolean> {
-        //TODO optimize
-        return msgs.map { list -> list.none {m -> !isValid(m)}}
+        return msgs.map { list -> list.none {m -> !validPredicate(m)}}
+    }
+
+    fun isNotValid(): Flow<Boolean> {
```

Comment:
a lazy property would be better imho, so there will be at most one instance of the mapped flow